src/coreclr/jit/codegencommon.cpp

```diff
@@ -6490,11 +7231,14 @@ void CodeGen::genReportGenericContextArg(regNumber initReg, bool* pInitRegZeroed
     // ARM's emitIns_R_R_I automatically uses the reserved register if necessary.
     GetEmitter()->emitIns_R_R_I(ins_Store(TYP_I_IMPL), EA_PTRSIZE, reg, genFramePointerReg(),
                                 compiler->lvaCachedGenericContextArgOffset());
-#else  // !ARM64 !ARM
+#elif defined(TARGET_LOONGARCH64)
+    genInstrWithConstant(ins_Store(TYP_I_IMPL), EA_PTRSIZE, reg, genFramePointerReg(),
+                         compiler->lvaCachedGenericContextArgOffset(), REG_R21);
```

Comment:
Is R21 a special "reserved register"? Maybe it should be `rsGetRsvdReg()` like ARM64?

Or maybe define something like `REG_LARGE_CONSTANT_TMP` for these purposes?

Reply:
> Is R21 a special "reserved register"? 

Yes, it is reserved.